acme/localStore.go

```diff
@@ -27,71 +21,105 @@ func NewLocalStore(file string) *LocalStore {
 	}
 }
 
-// Get atomically a struct from the file storage
-func (s *LocalStore) Get() cluster.Object {
-	s.storageLock.RLock()
-	defer s.storageLock.RUnlock()
-	return s.account
-}
-
-// Load loads file into store
-func (s *LocalStore) Load() (cluster.Object, error) {
-	s.storageLock.Lock()
-	defer s.storageLock.Unlock()
+// Get loads file into store and returns the Account
+func (s *LocalStore) Get() (*Account, error) {
 	account := &Account{}
 
-	err := checkPermissions(s.file)
+	hasData, err := checkFile(s.file)
 	if err != nil {
 		return nil, err
 	}
-	f, err := os.Open(s.file)
-	if err != nil {
-		return nil, err
-	}
-	defer f.Close()
-	file, err := ioutil.ReadAll(f)
-	if err != nil {
-		return nil, err
-	}
-	if err := json.Unmarshal(file, &account); err != nil {
-		return nil, err
+
+	if hasData {
+		f, err := os.Open(s.file)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		file, err := ioutil.ReadAll(f)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal(file, &account); err != nil {
+			return nil, err
+		}
 	}
-	account.Init()
-	s.account = account
-	log.Infof("Loaded ACME config from store %s", s.file)
 	return account, nil
 }
 
-// Begin creates a transaction with the KV store.
-func (s *LocalStore) Begin() (cluster.Transaction, cluster.Object, error) {
-	s.storageLock.Lock()
-	return &localTransaction{LocalStore: s}, s.account, nil
-}
+// ConvertToNewFormat converts old acme.json format to the new one and store the result into the file (used for the backward compatibility)
+func ConvertToNewFormat(fileName string) {
+	localStore := acme.NewLocalStore(fileName)
+	storeAccount, err := localStore.GetAccount()
+	if err != nil {
+		log.Warnf("Failed to read new account, ACME data conversion is not available : %v", err)
```

Comment:
Is the new account being read, or written?

Reply:
@dtomcej Here it's the new account.
If the value is `nil` we assume the `acme.json` file is in the old format and has to be converted but if `err != nil` we stop the conversion.